### DISCUSSION TITLE
should vision by zero contain the signs of divisor and dividend?

### DISCUSSION THREAD

**Discussion by inoas** [gleam-lang/gleam#3012] [Ideas & suggestions]
Given this notice about OTP 27: 

https://erlangforums.com/t/in-erlang-otp-27-0-0-will-no-longer-be-exactly-equal-to-0-0/2586/1

> For that we have the exact equality operator (=:=) which returns whether two terms are indistinguishable. That is for any terms X and Y, X =:= Y returns whether f(X) =:= f(Y) for all pure functions f.

Should gleam's division by zero not also keep the sign?

```gleam
import gleam/io

pub fn main() {
   1 /  0 |> io.debug //  0
  -1 / -0 |> io.debug //  0
   1 / -0 |> io.debug // -0 but says 0
  -1 /  0 |> io.debug // -0 but says 0
}
```

**Comment by lpil** [comment]
Thank you, but this would be a breaking change, and now that we are v1 we don't have breaking changes.

**Reply by inoas-nbw** [reply]
Maybe this is something for Gleam 2.0?

**Reply by inoas-nbw** [reply]
And is it certain that the current behavior is not a bug.. what does pony do?

**Reply by inoas-nbw** [reply]
```pony
actor Main
  new create(env: Env) =>
    env.out.print((I64(1) / I64(-0)).string())
```
prints 0 not -0. So gleam is in line with that impl.
